backend/python/app/rest/entity_routes.py

```diff
@@ -169,7 +169,7 @@ def delete_entity(id):
         error_message = getattr(e, "message", None)
         return jsonify({"error": (error_message if error_message else str(e))}), 500
 
-    return jsonify(result), 200
+    return jsonify(result), 204
```

Comment:
200 makes more sense because we're returning content here (ID of the deleted entity)

> I remember discussing the response code for delete before but I can't remember why we use 200 instead of 204, was there a reason?

I think we actually want to converge on always returning 200. GraphQL best practice is to always return a value from queries/mutations so for parity, REST should also return a value (only some(?) of our GraphQL delete mutations are currently returning values, but it's something we should fix 😅)